src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -891,9 +891,8 @@ private BoundCollectionExpression ConvertCollectionExpression(
                 }
 
                 implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
-                collectionCreation = BindCollectionExpressionConstructor(syntax, targetType, constructor, diagnostics);
-                Debug.Assert((collectionCreation is BoundNewT && !isExpanded && constructor is null) ||
-                             (collectionCreation is BoundObjectCreationExpression creation && creation.Expanded == isExpanded && creation.Constructor == constructor));
+                collectionCreation = BindCollectionExpressionConstructor(syntax, targetType, constructor, node.WithElement, diagnostics);
+                Debug.Assert(collectionCreation is BoundObjectCreationExpressionBase or BoundBadExpression);
```

Comment:
tests are not done as BindCollectionExpressionConstructor can bind to different constructors now.

Reply:
sorry let me clarify.  I'm saying that the prior tests were changed as they are no longer valid in the world where there can be multiple constructor candidates, and that overload resolution is now used to pick the right one to call.  I was trying to explain why the assert went away.